src/pac.c

```diff
@@ -435,6 +435,43 @@ pac_stash_decayed(tsdn_t *tsdn, pac_t *pac, ecache_t *ecache,
 	return nstashed;
 }
 
+static bool
+decay_with_process_madvise(edata_list_inactive_t *decay_extents) {
+	cassert(have_process_madvise);
+	assert(opt_process_madvise_max_batch > 0);
+#ifndef JEMALLOC_HAVE_PROCESS_MADVISE
+	return true;
+#else
+	size_t max_stack = VARIABLE_ARRAY_SIZE_MAX / sizeof(struct iovec);
+	size_t len = max_stack < opt_process_madvise_max_batch ? max_stack :
+	    opt_process_madvise_max_batch;
+	struct iovec vec[len];
+	size_t cur = 0, total_bytes = 0;
+	for (edata_t *edata = edata_list_inactive_first(decay_extents);
+	     edata != NULL;
+	     edata = edata_list_inactive_next(decay_extents, edata)) {
+		size_t pages_bytes = edata_size_get(edata);
```

Comment:
we don't remove things away from the list. In case `pages_purge_process_madvise` failed after some success, these items will be repeatedly purged? Does this matter? Or we are doing this just to save time?

Reply:
If pmadvise is partial success, yes some regions will be purged multiple times but there will be no correctness concern. I'm not too worried about that case, because it's not suppose to happen unless there's heap corruption.

Reply:
Agreed. As long as purging multiple times does not have correctness concern, we should be good. Maybe add a comment to explain this just for future references?

Reply:
Agree. Added comments around `already_purged`